### ISSUE TITLE
Add Roland SC-55 emulation

### ISSUE THREAD

**Issue by rderooy** [dosbox-staging/dosbox-staging#2138]
### Are you using the latest Dosbox-Staging Version?

- [X] I have checked releases and am using the latest release.

### Different version than latest?

latest git

### What Operating System are you using?

Linux x86_64

### Is your feature request related to a problem? Please describe.

Add Roland SC-55 emulation by use of the [emusc](https://github.com/skjelten/emusc) library

Many DOS games with General-MIDI music were developed on the SC-55, and some even specifically list the Roland SoundCanvas in the music options. Some even support the Roland GS specific extensions.

Right now emusc only supports the SC-55 and SCC-1 ROMs. But the developer has indicated possibly adding the SC-88 in the future.

### Describe alternatives you've considered

I have successfully used emusc on Linux with dosbox-staging with `mididevice=alsa`. It automatically picked up the emusc midi device and playing back MIDI worked. Still having it integrated, ala FluidSynth would be easier.

![image](https://user-images.githubusercontent.com/17701098/208252466-97d45d88-fb12-4eb1-9946-666a5e30003a.png)

### Code of Conduct & Contributing Guidelines

- [X] Yes, I agree.

**Comment by Grounded0** [comment]
We're already actively monitoring the project progress by @skjelten and others and likely will incorporate it once it matures a bit more.

**Comment by johnnovak** [comment]
Agree with @Grounded0 ; we should only incorporate this into DOSBox once they've hit at least 80-90% of fidelity, otherwise it's pointless and confusing to the users. (Plus there's a non-zero risk that the project will be abandoned without ever reaching full fidelity, in which case we'd need to remove it if we don't want to work on it ourselves.)

I'd say we should wait.

**Comment by grapeli** [comment]
It's still too early. Version 0.0.1 clearly indicates this.

It's easy to convince yourself that not everything sounds as it should.
Theme from the main menu of Ultima VII Part 2 - Serpent Isle. It's better not to play too loud, especially with headphones on.

https://user-images.githubusercontent.com/452325/208630641-347b5c73-6e51-46f6-8e18-209f23a82cf2.mp4

The introduction to Ultima VI - The False Prophet sounds great to me.

[Ultima VI - The False Prophet Introduction.webm](https://user-images.githubusercontent.com/452325/208630688-26fd7ee4-6cd2-4760-853f-1c92951a3866.webm)

**Comment by johnnovak** [comment]
> The introduction to Ultima VI - The False Prophet sounds great to me.

It's terribly off. Just compare it to this real hardware recording:

https://www.youtube.com/watch?v=yv__aZzmXBA

Yes, the music during the intro anim sounds "pleasant", but it's waaaay off. Then the game menu music is a complete disaster, sounds like the filter emulation is completely missing, half the notes are out of tune, and so on.

Sure, you gotta start somewhere, so this is fine for a 0.0.1 release, don't get me wrong. I really hope they will develop this into a fine SC-55 emulation library, given enough time. I'm only saying it's very far away from being ready to be incorporated into our codebase.

**Comment by grapeli** [comment]
@johnnovak 
I am satisfied with the sound of the introduction to this game obtained from emusc.
I agree with the opinion regarding the sound in the main menu.

I would be careful in comparing this "real" sound obtained on the original equipment. I really don't know what other devices the signal passed through before it was finally recorded. I don't have such equipment.

The same objections can be made to the MT-32 version.
The sound obtained in dosbox-staging thanks to mt-32 emulation is also very far from such an "[original](https://www.youtube.com/watch?v=bElNUSvoXxE)".

**Comment by johnnovak** [comment]
Nuked-SC55 is the most likely contender now, it already sounds awesome. @Grounded0, rejoice 😎 

https://www.youtube.com/watch?v=u4Z5y2otJqY

More info:
https://www.vogons.org/viewtopic.php?f=24&t=99447

> this is continuation of my early sc-55 emulation attempt I started in 2021. The basic idea was to emulate all main chips in proper way and use original SC-55 firmware as is (MAME approach). Back then I made emulator of its main CPU but was not able to properly boot firmware on it, and combined with lack of information about PCM chip I ultimately abandoned it.
Earlier this year though I decided to try tinker with this a bit more, and tried to trace die shot of decapped pcm chip (from John McMaster/siliconpr0n.org). After completing tracing I used org/ogamespec's deroute utility to convert netlist to verilog code and I started analyzing it. Analysis revealed a lot of useful info on how synthesis engine works, and how PCM chips is interfaced by CPU. Using this new info I revived my old code and started fixing it. Eventually I was able to boot firmware and then started adding PCM chip emulation. At the moment PCM chip emulation is halfway done and it already sounds awesome. Now stuff like TVF, chorus/reverb/mixing are left.

**Comment by rderooy** [comment]
It would be great if he could work together with the developer of emusc...

**Comment by shermp** [comment]
> It would be great if he could work together with the developer of emusc...

They are taking completely different emulation approaches. emusc is taking the MUNT approach.

**Comment by Burrito78** [comment]
ROMs are here, thanks @NicknineTheEagle!
https://archive.org/download/mame-merged/mame-merged/sc55mk2.zip

**Comment by Grounded0** [comment]
![image](https://github.com/dosbox-staging/dosbox-staging/assets/48192273/71fe45b7-33bc-471d-b27e-d37514eb3f1e)

**Comment by Grounded0** [comment]
Nuked SC55 emulator - demo songs:

https://www.youtube.com/watch?v=DUEkcFTuldg

**Comment by MasterO2** [comment]
> Nuked SC55 emulator - demo songs:
> 
> https://www.youtube.com/watch?v=DUEkcFTuldg

Once Nuked SC55 emulator is finished, it will make a great addition to Staging!

**Comment by shermp** [comment]
@weirddan455 noticed that the license might be an issue (original MAME license which is not compatible with GPL).

**Comment by johnnovak** [comment]
> @FeralChild64 noticed that the license might be an issue (original MAME license which is not compatible with GPL).

Our Voodoo code originates from MAME. Isn't it the same deal @shermp ?

**Comment by shermp** [comment]
It looks like the voodoo code we used is BSD licensed.

**Comment by Grounded0** [comment]
We have to ask the author is the person willing to dual licence later.

**Comment by johnnovak** [comment]
We'll bribe him or use whatever means necessary to release it under a license everyone is happy with 😎 I'd hate to be able to use this only in my personal builds...

0.1 was released 9 hours ago!!!! 🎉 So excited, I'm gonna build it and compare it with my hardware recordings!

**Comment by MasterO2** [comment]
> We'll bribe him or use whatever means necessary to release it under a license everyone is happy with 😎 I'd hate to be able to use this only in my personal builds...
> 
> 0.1 was released 9 hours ago!!!! 🎉 So excited, I'm gonna build it and compare it with my hardware recordings!

Blackjack and hookers. 😄

**Comment by nickkeane** [comment]
@johnnovak Would it be possible to integrate an optional overlay to show something like an MT-32 Panel or SC-55 MCU at some point? I use fullscreen maintain aspect integer scaling so I have a lot of black border space to the sides of the DOSBox window. Would be nice to add an actual border overlay background as well.

**Comment by johnnovak** [comment]
> @johnnovak Would it be possible to integrate an optional overlay to show something like an MT-32 Panel or SC-55 Display at some point? I use fullscreen maintain aspect integer scaling so I get a lot of black border space to the sides of the DOSBox window. Would be nice to add an actual border overlay background as well.

Yes, such things are planned in the upcoming OSD work:
https://github.com/dosbox-staging/dosbox-staging/discussions/1953

The screenshot below is a mockup for displaying the special hidden MT-32 messages some games send to the MT-32's LCD. But support could be extended to show the MT-32 and SC-55 LCDs all the time. It would be a super cool feature, and I will do it when I get to it, yeah.

https://user-images.githubusercontent.com/698770/191507008-e2eaa42a-f7fe-4f76-a021-80d06690d447.png

**Comment by Grounded0** [comment]
We got clarification on the licence issue:

https://github.com/nukeykt/Nuked-SC55/issues/13#issuecomment-2035247178

**Comment by MasterO2** [comment]
So, as long as `Nuked-SC55` uses that license , will Staging not implement SC-55 emulation as a result?

**Comment by Grounded0** [comment]
@MasterO2 

No. But the author may be open to dual licencing or changing the licence as the person stated.

**Comment by Grounded0** [comment]
More about dual licencing:

https://en.wikipedia.org/wiki/Multi-licensing

**Comment by NicknineTheEagle** [comment]
> nuked-sc55 requires one rom that's not included in mame romset (or at least wasn't last time I looked) and is challenging to dump yourself. he had to voltage glitch to do it.
> 
> and it's pretty well done already. people ripping it off to sell fake sound canvas modules is not an idle concern, and he seems to want to make it very clear to Roland that they get to sue people who do that, and that's it not his fault.
> 
> It works pretty well attaching with a virtual midi cable already.

All necessary ROMs are currently included in MAME set.

**Comment by FeralChild64** [comment]
Why Roland does not sell these old ROMs (via GOG, for example) for emulation purposes is beyond me.

**Comment by johnnovak** [comment]
As @FeralChild64 mentioned it at Discord the other day, there is probably a way around this.

- Implement a very simple MIDI device plugin system: host can send MIDI commands, plugin can fill a buffer with audio data.
- We release the plugin interface under a BSD license that's compatible with anything.
- We create a second StagingSC55MidiPlugin project or whatever that implements our plugin interface and has the NukedSC55 code. This could be released under MAME.
- Users will need to click one more button to download the plugin DLL file, or maybe we could auto-download at startup. We won't be violating the GPL as this is an optional component and as we're only communicating with this component over an abstract interface, it's irrelevant what license the plugin has from Staging's perspective.

The upside is that this is the least risky way for Staging to add direct support to NukedSC55.

**Comment by zaphod77** [comment]
sounds like a good plan.

> > nuked-sc55 requires one rom that's not included in mame romset (or at least wasn't last time I looked) and is challenging to dump yourself. he had to voltage glitch to do it.
> 
> All necessary ROMs are currently included in MAME set.

ahh, the sc55mk2.zip that currently requires a password to download. got it.

**Comment by NicknineTheEagle** [comment]
> sounds like a good plan.
> 
> > > nuked-sc55 requires one rom that's not included in mame romset (or at least wasn't last time I looked) and is challenging to dump yourself. he had to voltage glitch to do it.
> > 
> > 
> > All necessary ROMs are currently included in MAME set.
> 
> ahh, the sc55mk2.zip that currently requires a password to download. got it.

I think you're looking at the wrong source. There's an up-to-date MAME set on Internet Archive and it includes sc55mk2.

**Comment by johnnovak** [comment]
Let's play guess the missing letters:

https://archive.🐨🐨🐨/download/mame-merged/mam🐨-merged/s🐨55m🐨2.zi🐨

😎

**Comment by Grounded0** [comment]
It was only added to the latest MAME release 0.264.

**Comment by Karmeck** [comment]
> > sounds like a good plan.
> > > > nuked-sc55 requires one rom that's not included in mame romset (or at least wasn't last time I looked) and is challenging to dump yourself. he had to voltage glitch to do it.
> > > 
> > > 
> > > All necessary ROMs are currently included in MAME set.
> > 
> > 
> > ahh, the sc55mk2.zip that currently requires a password to download. got it.
> 
> I think you're looking at the wrong source. There's an up-to-date MAME set on Internet Archive and it includes sc55mk2.

so why did you remove the link? can still be seen in history, but anyway. curious.

**Comment by Grounded0** [comment]
Yeah we can say it was added to MAME release 0.264. It's up to people to find it. Not our problem.

**Comment by Grounded0** [comment]
EmuSC is reaching its initial release in the next few weeks so that should be looked into too. If Roland lawyers don't care then maybe we're overreacting a bit. It's LGPL 3+ for libEmuSC and GPL 3+ for EmuSC so should be interesting.

**Comment by johnnovak** [comment]
Yeah, give it 3 months, then integrate 🤷🏻

**Comment by zaphod77** [comment]
i actually found the right rom file, but googling didn't work.

**Comment by johnnovak** [comment]
LOL, so we're back to the times when we're treating ROM dumps like some super secret alien technology... Okay... 😆We had that with the MT-32 ROMs...

**Comment by Grounded0** [comment]
Problem with the MT-32 was that the case was already settled in 2003 but the secrecy continued:

https://yro.slashdot.org/story/03/12/14/2329244/roland-backs-down-on-mt-32-emulator

**Comment by NicknineTheEagle** [comment]
> Problem with the MT-32 was that the case was already settled in 2003 yet the secrecy continued:
> 
> https://yro.slashdot.org/story/03/12/14/2329244/roland-backs-down-on-mt-32-emulator
> 
> We haven't had clear indication on the SC-55 yet but the initial release of EmuSC should reveal it.

I've read into it and the only reason Roland backed is because they couldn't find documentation confirming their ownership of firmware ROM. Which is hilarious but doesn't quite make the case "settled".

**Comment by Grounded0** [comment]
I feel sorry for the intern that is still looking for documentation confirming their ownership of firmware ROM 20 years later.

**Comment by johnnovak** [comment]
Yeah, but reality check, what's in it for them for fucking around with free emulator projects? There's no money in these projects.

These emulators are not causing them any monetary loss. Yet if it becomes a scandal that a big corporation is threatening free and open-source emu projects, that's definitely a loss of prestige for them which _can_ result in monetary loss if certain groups of people boycott them.

But even then, they're a musical instrument company still, they're not in the computer game business for a long time. So the only rational thing to do for them is basically not to give a shit. Only retro-gamers are interested in emulating the SC-55, objectively it has next to zero current day usage. No one would buy a "remade SC-55" hardware for making music.

**Comment by johnnovak** [comment]
> > Problem with the MT-32 was that the case was already settled in 2003 yet the secrecy continued:
> > https://yro.slashdot.org/story/03/12/14/2329244/roland-backs-down-on-mt-32-emulator
> > We haven't had clear indication on the SC-55 yet but the initial release of EmuSC should reveal it.
> 
> I've read into it and the only reason Roland backed down is because they couldn't find documentation confirming their ownership of firmware ROM. Which is hilarious but doesn't quite make the case "settled".

I think they realised what I said above. No monetary gain is involved by going in all-guns-blazing "protecting" their creation no one except retro-games care about, but there's a potential for a lot of prestige damage if it blows up. Big corporation going after the broke open source devs, that doesn't make pretty headlines.

**Comment by MasterO2** [comment]
> > > Problem with the MT-32 was that the case was already settled in 2003 yet the secrecy continued:
> > > https://yro.slashdot.org/story/03/12/14/2329244/roland-backs-down-on-mt-32-emulator
> > > We haven't had clear indication on the SC-55 yet but the initial release of EmuSC should reveal it.
> > 
> > 
> > I've read into it and the only reason Roland backed down is because they couldn't find documentation confirming their ownership of firmware ROM. Which is hilarious but doesn't quite make the case "settled".
> 
> I think they realised what I said above. No monetary gain is involved by going in all-guns-blazing "protecting" their creation no one except retro-games care about, but there's a potential for a lot of prestige damage if it blows up. Big corporation going after broke open source devs—that doesn't make pretty headlines.

Space Quest 7: Rolando Strikes Back  😄

**Comment by Grounded0** [comment]
Berne Convention became effective 5 December **1887**:

https://en.wikipedia.org/wiki/Berne_Convention

**Comment by johnnovak** [comment]
> Berne Convention became effective 5 December **1887**:
> 
> https://en.wikipedia.org/wiki/Berne_Convention
> 
> There's no point arguing since EmuSC is about to ship. We'll see.

About to ship? They're very far away from achieving even just OK results...

**Comment by Grounded0** [comment]
But ship it will quite soon:

https://github.com/skjelten/emusc/issues/40#issuecomment-1992693752

**Comment by johnnovak** [comment]
> But ship it will quite soon:
> 
> [skjelten/emusc#40 (comment)](https://github.com/skjelten/emusc/issues/40#issuecomment-1992693752)

You must be referring to this sentence, but that's not my reading of it:

> So, to answer your question, I guess it will be fair to say that the project is about to transition from "an early development stage" to "half the way there". 

He's just saying things are "halfway there", but nothing about any actual upcoming release.

**Comment by NicknineTheEagle** [comment]
> Berne Convention became effective 5 December **1887**:
> 
> https://en.wikipedia.org/wiki/Berne_Convention
> 
> There's no point arguing since EmuSC is about to ship. We'll see.

@zaphod77 is talking about USA where Berne Convention was not adopted until 1989.

**Comment by johnnovak** [comment]
> Here's more:
> 
> [skjelten/emusc#32 (comment)](https://github.com/skjelten/emusc/issues/32#issuecomment-1987177915)

Ok, that shows some actual intent 😄

**Comment by Grounded0** [comment]
Yeah couldn't remember where i saw that stronger affirmation first but there you go.

**Comment by johnnovak** [comment]
> Why Roland does not sell these old ROMs (via GOG, for example) for emulation purposes is beyond me.

Yeah, sell them for 10 bucks, everybody would legally buy them, and job done 🤷🏻

**Comment by FeralChild64** [comment]
More like "Spend 10 USD (or even 25 USD) and the following games from your GOG library will have an enhanced music version, here are some samples, just listen".

BTW, People still purchase JiffyDOS for Commodore, Kickstart&Workbench for Amiga, why there is no way to purchase old Roland ROMs?

**Comment by johnnovak** [comment]
People download JiffyDOS & the Kickstart pack and call it a day 😄 

Btw, I don't think Roland will give a fuck about this. The SCVA is basically a failed experiment, they haven't updated it in like 3 years. They'll just let it die, and that's it.

**Comment by zaphod77** [comment]
Roland is very much aware of the desire for people to play old games. that's why there's  daughterboard with built in roland samples on it that is licensed.  With a license preventing you from selling it as a stand alone tone generator or using it as such in public.

they do also still sell lifetime keys for Sound Canvas VA. (it's 69$ on a number of other websites, you just need a free cloud account to use it, which gives you just "Zenology Lite") . While they aren't updating it, because there are no new sound canvases being made, they have no reason to discontinue it either.

That said, here's a project that would be useful and 100% legal.

Make a replacement of the ms gs software synth, that is fully sysex and NRPN compatible with a sc55mkII.  It will load the DLS file that comes with windows. Since that DLS file is licensed for use under ms operating systems, no roms are needed.

That GM.DLS file is under the following license.
C:\Windows\System32\drivers>more gmreadme.txt
-------------------------------------------------------------------------
GMREADME.TXT
Copyright (c) 1998-2000 Microsoft Corporation.  All Rights Reserved.
------------

The GM.DLS file contains the Roland SoundCanvas Sound Set which is
protected under the following copyright:
Roland GS Sound Set/Microsoft (P) 1996 Roland Corporation U.S.
The Roland SoundCanvas Sound Set is licensed under Microsoft's
End User License Agreement _for use with Microsoft operating
system products only_.  All other uses require a separate written
license from Roland.

-------------------------------------------------------------------------

Now i'm not a lawyer, but i think loading this sample data into a more capable softsynth, provided the file is found in the windows system32\drivers\ directory, is almost certainly allowed, depending on what the legal definition of a "Microsoft operating system product" is.  This is supported by me having asked Roland back in the day if it was okay for me to try to convert it to .sf2 to use on my awe32, and them saying they don't care if it's done for personal use, as the samples are low quality, but not to upload it.

**Comment by FeralChild64** [comment]
> People download JiffyDOS & the Kickstart pack and call it a day 

Well, looking at some recent sh*tstorms, I suppose a lot of our users lack the technical expertise to reconfigure both the DOSBox and the game by themselves :D BTW, I tend to buy such packages if they are relevant for retro systems within my interests.

**Comment by Grounded0** [comment]
I used to use Roland licenced integrated softsynth on macOS and its a no go due to its poor quality 8-bit samples. Roland softsynth integrated into Windows and macOS is a subpar product and the soundfont is 1.5 MB in size on macOS.

**Comment by johnnovak** [comment]
> I used to use Roland licenced integrated soundfont on macOS and its a no go due to its poor quality 8-bit samples. Roland soundfont integrated into Windows and macOS is a subpar product and the soundfont is 1.5 MB in size on macOS which is not enough for quality samples. Thats why i recommend 4GMGSMT because its essentially the same but with 16-bit 44.1 kHz samples.

Agreed.

Maybe we just all calm down a bit and wait and see before come up with ridiculous overengineered solutions like plugin systems, ripping OS-included samples and shit 😎 

If after 6+ months this ends up in several emulators and projects and Roland doesn't give a shit, we'll just persuade the guy to dual license it or something and we integrate it into Staging.

Until then, people can learn how to use a loopback MIDI cable (and yeah, macOS users are out of luck).

**Comment by Grounded0** [comment]
FluidSynth has gained undeservedly bad reputation due to clueless users being recommended to use garbage soundfonts with it. I lose my shit when i hear instruments striking at wrong amplitude against each other. That's what you get for actually using a wavetable card through the 1990s unlike people who didn't give a damn or were short on money.

**Comment by johnnovak** [comment]
> FluidSynth has gained undeservedly bad reputation due to clueless users being recommended to use garbage soundfonts with it. I lose my shit when i hear instruments striking at wrong amplitude against each other. That's what you get for actually using a wavetable card through the 1990s unlike people who didn't give a damn or were short on money.

Yeah, use the 4MB creative one, or GeneralUser GS and even Fluid R3 are good alternative options. The rest is rather meh.

**Comment by zaphod77** [comment]
4MBgmgsmt is NOTHING like the sound canvas. never has been. Its square wave, for example, isn't very sound canvasy, and it doesn't even have the mkii extra instruments, if it's the one i remember. i honestly prefer gm.dls to it.  Quicktime's version is horrible, though.

there are some decent sound canvas soundfonts, and someone finally did a real sample rom rip and made it into one. it's close enough for most games, though good luck playing back right in the night gs properly on it.

in the meantime, i will use nuked sc55 and a virtual midi cable until it gets to the point where staging can link to it without one. roland has pretty much given their approval for use of emulation and their samples to play old pc games, but they will act if you go beyond this goal.

Any soundfont that is truly GS compliant will play back stuff for sound canvas with appropriate dynamics.  Most "GS" soundfonts, well, don't.

The creative/emu fonts for the awe32 don't cut it.  Creative wavesynth doesn't cut it.  

Fluidsynth lite is janky as heck, and doesn't even have portmanteau. the full fluidsynth does, but it's still rather janky.

**Comment by Grounded0** [comment]
The main problem i see with most soundfonts is the amateur hour stuff like instruments striking at wrong velocities to each other when it should be extremely neutral unless differing note velocity is specified. Not having a balanced soundscape kills it for me. FluidSynth can't fix crap soundfonts.

**Comment by johnnovak** [comment]
@zaphod77 You're preaching to the choir, man 😄 We all want NukedSC55, it's dead on perfect.

All we're saying FluidSynth is not too bad in the interim with the proper SoundFonts, and while I'm super into music and appreciate subtleties you listed, many people frankly don't care. There is music, it's 80% there, OK, let's play the game. So the SoundFonts we listed are fine for them.

Are they same as the SC-55? No way. Are they well-balanced, unlike 99% of amateur SF2 files? Big resounding yes. They make listening to GM/GS music enjoyable, while not being 100% true to the original SC-55 rendition.

**Comment by zaphod77** [comment]
I know for a fact that they will get pissed if you try to make a standalone tone generator, because the license on the X2GS specifically prohibits making it into an instrument. this is also what nuked is trying to enforce with the current licensing.

**Comment by johnnovak** [comment]
> I know for a fact that they will get pissed if you try to make a standalone tone generator, because the license on the X2GS specifically prohibits making it into an instrument. this is also what nuked is trying to enforce with the current licensing.

Since when is DOSBox Staging an instrument? Hmm? 😄

**Comment by zaphod77** [comment]
I know it's not. which is why i don't think roland will really have an issue with it being in dosbox somehow.

**Comment by Falcosoft** [comment]
> I used to use Roland licenced integrated soundfont on macOS and its a no go due to its poor quality 8-bit samples. Roland soundfont integrated into Windows and macOS is a subpar product and the soundfont is 1.5 MB in size on macOS which is not enough for quality samples. That's why i recommend 4GMGSMT because its essentially the same but with 16-bit 44.1 kHz samples.

While I agree that 4GMGSMT.SF2/CT4MGM.SF2 (exactly the same one) is a decent 4MB SF2 soundfont but be aware that all Creative/Emu made SF2 soundfonts have rather serious bugs (such as instrument range definition, base pitch definition etc.)
E.g. just listen to Storm.mid from Duke Nukem3D using 4GMGSMT.SF2. The cricket mimicking effect is completely missing at the beginning:
[STORM.zip](https://github.com/dosbox-staging/dosbox-staging/files/14893198/STORM.zip)

**Comment by Burrito78** [comment]
@Falcosoft Just out of curiosity, what's your favorite universal (e.g. suited for Duke 3D, Doom and TIE Fighter or Sam and Max) "go to" SF2 soundfont?

**Comment by Grounded0** [comment]
I'm also interested of hearing about better alternatives to it. We're always open to tweaking our list at:

https://github.com/dosbox-staging/dosbox-staging/wiki/MIDI#soundfonts

**Comment by Falcosoft** [comment]
> @Falcosoft Just out of curiosity, what's your favorite universal (e.g. suited for Duke 3D, Doom and TIE Fighter or Sam and Max) "go to" SF2 soundfont?

My shameless recommendation is Reality_GMGS_falcomod.sf2 :)
[http://falcosoft.hu/reality_gmgs_falcomod.zip](http://falcosoft.hu/reality_gmgs_falcomod.zip)
With some VST effects it can sound rather good with Doom:
[https://youtu.be/WrCKxR3jhrg?si=1t65W5ESAOR5oXkd](https://youtu.be/WrCKxR3jhrg?si=1t65W5ESAOR5oXkd)
But the recommended soundfonts like GeneralUser GS or FluidR3 are good choices as well.
If you know the limitations of Creative/Emu made soundfonts they can be also good in 95% of cases.
The big exception is 28MBGM.sf2 from the X-Fi era that has one of the biggest bugs ever made in an 'officially' released soundfont by Creative/Emu. Namely it completely misses Picked Bass (34) preset from the capital tones. So its usage is absolutely not recommended.
Otherwise typical problem with Creative/Emu soundfonts that the last 1.5 ovtave (keys 109-127) is never defined in case of melodic instruments and in some cases the last 3 ovtaves is missing. It can be argued that these ranges are rarely "musical" enough but they can be (and actually were) used for effects. Moreover Yamaha and Roland gears always produce at least high pitched "something" in these ranges.
8MBGMSFX.SF2/CT8MGM.SF2 contains no GS/MT-32 variation tones at all (137 presets overall but at least it has fully implemented capital tones :Picked Bass is not missing).
In case of 4GMGSMT.SF2/CT4MGM.SF2 (full GS variation tones and Drum sets + MT-32 tones and Drum set: 329 presets overall.) some of the GS effects (Machine Gun, Explosion etc.) have wrong base note. They are misplaced by 2 octaves lower.

**Comment by Cacodemon345** [comment]
By the way, for emusc: https://github.com/skjelten/emusc/issues/45

The libemusc library is not licensed under LGPLv2+ at the moment, but LGPLv3+. DOSBox Staging uses a GPLv2+ license.

I'm aware of skjelten's stance on that library at the moment, which is that it will be only looked into if any projects express interest on a LGPLv2+, but I'm not aware if this project is willing to do so at the moment.

**Comment by johnnovak** [comment]
> I'm aware of skjelten's stance on that library at the moment, which is that it will be only looked into if any projects express interest on a LGPLv2+, but I'm not aware if this project is willing to do so at the moment.

We are very much interested to integrate it when it's at least 70-80% fidelity. Now, it's perhaps 20-30%.

Having said that, we don't need _two_ SC-55 implementations in DOSBox, we need only one. NukedSC55 is definitely our preferred options, and it will go nicely with NukedOPL which is our OnlyTrueOPLEmulation option (we nuked the rest—I'll show myself out...)

**Comment by Karmeck** [comment]
we are sharing soundfonts recommendations?

SC-55.SoundFont.v1.3b

I Contribute with this one:
https://www.doomworld.com/forum/topic/118828-trevor0402s-sc-55-soundfont/

Download
https://www.doomworld.com/forum/post/2493425

**Comment by Burrito78** [comment]
> we are sharing soundfonts recommendations?
> 
> SC-55.SoundFont.v1.3b
> 
> I Contribute with this one: https://www.doomworld.com/forum/topic/118828-trevor0402s-sc-55-soundfont/
> 
> Download https://www.doomworld.com/forum/post/2493425

That's the most authentic SC-55 SF2 for Staging right now, did anyone test it yet, thanks!

**Comment by Grounded0** [comment]
EmuSC went for its initial release v0.1.0:

https://github.com/skjelten/emusc/releases/tag/v0.1.0

**Comment by Grounded0** [comment]
EmuSC roadmap:

https://github.com/skjelten/emusc/wiki/Roadmap

**Comment by johnnovak** [comment]
**Nuked SC-55**

**Staging 0.82**

**This year**

Watch me 😎

**Comment by Karmeck** [comment]
> **Nuked SC-55**
> 
> **Staging 0.82**
> 
> **This year**
> 
> Watch me 😎

As long as nukeykt is okay with it, there should be no issue.

**Comment by johnnovak** [comment]
> As long as nukeykt is okay with it, there should be no issue.

I am going to implement it as a plugin that Staging can load to get over the licensing incompatibility between the MAME license of NukedSC55 and GPL2 of Staging.

My NukedSC55 plugin, living in a separate repo, will be under the NukedSC55 MAME license, of course.

Nukey doesn't need to be "okay with it" as I'm not going to violate his license of choosing in any shape or form. As long as forks and spinoff projects carry forward his license and adhere to its term, nobody need to ask for permission.

Of course, I hope he will like and appreciate my work, but that's a separate thing 😄

**Comment by MasterO2** [comment]
https://nerdlypleasures.blogspot.com/2024/06/true-roland-sc-55-emulation-is-here.html

**Comment by johnnovak** [comment]
> @johnnovak What is the current status of the NukedSC55 plugin?

Haven't started on it yet. Planning to do it in the next few months.

**Comment by johnnovak** [comment]
> [Roland RAP-10](https://www.mobygames.com/attributes/attribute/87/) has 185 games listed at Mobygames - [ISA card](https://en.wikipedia.org/wiki/Roland_Sound_Canvas#Sound_Canvas_Personal_Computer_Products), includes two-channel 16-bit digital audio support.
> 
> Does it make sense to emulate that as well - or having SC-55 in addition to the cards already emulated will cover it?

We will emulate all models NukedSC55 can emulate and is applicable for DOS games.

**Comment by zaphod77** [comment]
The RAP 10 is essentially a sound canvas without any of the variation instruments.  while emulation of it is probably possible, it's rather pointless when you can juts emulate a gm compatible sc-55.

**Comment by johnnovak** [comment]
Done in [#4090](https://github.com/dosbox-staging/dosbox-staging/pull/4090)

**Comment by zaphod77** [comment]
Nuked-sc55 has now been relicensed to gpl 2.0 or later. so it can now be directly implemented.  :)

**Comment by johnnovak** [comment]
> Nuked-sc55 has now been relicensed to gpl 2.0 or later. so it can now be directly implemented. :)

We'll keep using the plugin.
